Userland/Services/LookupServer/MulticastDNS.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2021, Sergey Bugaev <bugaevc@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include "DNSAnswer.h"
+#include "DNSName.h"
+#include "DNSPacket.h"
+#include <AK/IPv4Address.h>
+#include <LibCore/UDPServer.h>
+#include <netinet/in.h>
+
+namespace LookupServer {
+
+class MulticastDNS : public Core::UDPServer {
+    C_OBJECT(MulticastDNS)
+public:
+    Vector<DNSAnswer> lookup(const DNSName&, unsigned short record_type);
+
+private:
+    explicit MulticastDNS(Object* parent = nullptr);
+
+    void announce();
+    ssize_t emit_packet(const DNSPacket&, const sockaddr_in* destination = nullptr);
+
+    void handle_packet();
+    void handle_query(const DNSPacket&);
+
+    Vector<IPv4Address> local_addresses() const;
+
+    DNSName m_hostname;
+
+    static constexpr sockaddr_in mdns_addr {
+        AF_INET,
+        // htons(5353),
+        0xe914,
+        { // 224.0.0.251
+            0xfb0000e0 },
```

Comment:
clang-format made me do this

Reply:
putting a trailing comma after the last entry should make it format correctly :)

Reply:
Would this work?
```suggestion
// 224.0.0.251
{ 0xfb0000e0 },
```

Reply:
Seems to work, thx!